bpf/tests/bpf_ct_tests.c

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: GPL-2.0
+/* Copyright (C) 2016-2020 Authors of Cilium */
+
+#include <bpf/ctx/skb.h>
+#include <bpf/api.h>
+
+/* most values taken from node_config.h */
+#define CONNTRACK
+#define ENDPOINTS_MAP test_cilium_lxc
+#define POLICY_PROG_MAP_SIZE ENDPOINTS_MAP_SIZE
+#define METRICS_MAP test_cilium_metrics
+#define NAT46_PREFIX {.addr = {                   \
+	0xbe, 0xef, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, \
+	0x0, 0x0, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0    \
+} }
+
+#define ENDPOINTS_MAP_SIZE 65536
+#define IPCACHE_MAP_SIZE 512000
+#define METRICS_MAP_SIZE 65536
+#define EVENTS_MAP test_events_map
+
+#define CT_MAP_TCP6 test_cilium_ct_tcp6_65535
+#define CT_MAP_ANY6 test_cilium_ct_any6_65535
+#define CT_MAP_TCP4 test_cilium_ct_tcp4_65535
+#define CT_MAP_ANY4 test_cilium_ct_any4_65535
+#define CT_MAP_SIZE_TCP 4096
+#define CT_MAP_SIZE_ANY 4096
+#define CT_CONNECTION_LIFETIME_TCP	21600
+#define CT_CONNECTION_LIFETIME_NONTCP	60
+#define CT_SERVICE_LIFETIME_TCP		21600
+#define CT_SERVICE_LIFETIME_NONTCP	60
+#define CT_SYN_TIMEOUT			60
+#define CT_CLOSE_TIMEOUT		10
+#define CT_REPORT_INTERVAL		5
+#define CT_REPORT_FLAGS			0xff
+#define MTU 1500
+
+#define DEBUG
+
+#include <lib/dbg.h>
+#include <lib/conntrack.h>
+#include <lib/conntrack_map.h>
+
+__section("action/test-nop")
+int test_nop(struct __ctx_buff *ctx)
+{
+	(void)ctx;
+	return CTX_ACT_OK;
+}
+
+__section("action/test-map")
+int test_map(struct __ctx_buff *ctx)
+{
+	struct ipv4_ct_tuple ct_key = (struct ipv4_ct_tuple){
+		.saddr = bpf_htonl(16843009), /* 1.1.1.1 */
+		.daddr = bpf_htonl(16843010), /* 1.1.1.2 */
+		.sport = bpf_htons(1001),
+		.dport = bpf_htons(1002),
+		.nexthdr = 0,
+		.flags = 0,
+	};
+
+	struct ct_entry ct_val = (struct ct_entry){
+		.rx_packets = 1000,
+		.tx_packets = 1000,
+	};
+	struct bpf_elf_map *ctmap = &CT_MAP_TCP4;
+
+	(void)ctx;
```

Comment:
Our convention for unused arguments is to use `__maybe_unused` which is defined in `include/bpf/compiler.h` and looks safe to pull in here.

Reply:
fixed!